### PR TITLE
feat: Add polygon networks to truffle and hardhat configs

### PR DESCRIPTION
**Pull request by nicholaspai**
Signed-off-by: Nick Pai <npai.nyc@gmail.com>

**Motivation**

Now that [infura](https://consensys.net/blog/press-release/infura-and-truffle-now-support-polygon/) supports Polygon, it makes adding the mainnet and testnet into our truffle config very easy.

**Summary**

Can now run truffle and hardhat commands like so:

- `yarn truffle migrate --network polygon-matic/mumbai`
- `yarn hardhat deploy --network matic/mumbai`

**Details**

This may be unnecessary for some PRs. Catch-all for detailed explanations about the implementation decisions and implications of the change.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [X]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/common/src/TruffleConfig.js [hunk 1]
```diff
@@ -176,6 +177,7 @@ addLocalNetwork(networks, "test");
 // Mainnet fork is just a local network with id 1 and a hardcoded gas limit because ganache has difficulty estimating gas on forks.
 // Note: this gas limit is the default ganache block gas limit.
 addLocalNetwork(networks, "mainnet-fork", { network_id: 1, gas: 6721975 });
+addLocalNetwork(networks, "polygon-matic-fork", { network_id: 137, gas: 6721975, gasPrice: 1000000000 });
```
Comment (mrice32) [inline packages/common/src/TruffleConfig.js:180]: Just out of curiosity, why is a hardcoded gas price required for the fork network?

File: packages/common/src/PublicNetworks.js [hunk 2]
```diff
@@ -43,9 +43,23 @@ const PublicNetworks = {
     daiAddress: "0xbF7A7169562078c96f0eC1A8aFD6aE50f12e5A99",
     wethAddress: "0xd0A1E359811322d97991E03f863a0C30C2cF029C"
   },
+  137: {
+    name: "polygon-matic",
+    etherscan: "https://explorer-mainnet.maticvigil.com/",
+    customTruffleConfig: {
+      confirmations: 2,
+      timeoutBlocks: 200,
+      gasPrice: 1000000000 // 1 gWei
```
Comment (mrice32) [inline packages/common/src/PublicNetworks.js:52]: If someone provides the `--gasPrice` parameter, this hardcoded value overrides that value... shouldn't it be the other way around? Seems like we might want to unify how the default and override behavior happens across the different networks.

File: packages/common/src/PublicNetworks.js [hunk 3]
```diff
@@ -43,9 +43,23 @@ const PublicNetworks = {
...
+  },
   80001: {
-    name: "mumbai",
-    etherscan: "https://explorer-mumbai.maticvigil.com/"
+    name: "polygon-mumbai",
+    etherscan: "https://explorer-mumbai.maticvigil.com/",
+    customTruffleConfig: {
+      confirmations: 2,
+      timeoutBlocks: 200,
+      gasPrice: 1000000000 // 1 gWei
```
Comment (mrice32) [inline packages/common/src/PublicNetworks.js:61]: minor nit: have you thought about just importing web3 at the top and using it to get 1 GWEI so this is easier to read (we do this in the truffle config to interpret the user input).

```js
const Web3 = require("web3");

...
gasPrice: Web3.utils.toWei("1", "gwei");
```

**Review by mrice32 — COMMENTED** [review]
Very nice! A few minor comments and nits.

File: packages/common/src/TruffleConfig.js (hunk 1 above)
Comment (nicholaspai) [reply packages/common/src/TruffleConfig.js:180]: Its not , we can remove!

Comment (nicholaspai) [reply packages/common/src/TruffleConfig.js:180]: I left it there on principle

File: packages/common/src/PublicNetworks.js (hunk 2 above)
Comment (nicholaspai) [reply packages/common/src/PublicNetworks.js:52]: Hmm interesting, the problem is that the default gas price of 20 on ETH is too high for Polygon which can work with 1

File: packages/common/src/TruffleConfig.js (hunk 1 above)
Comment (mrice32) [reply packages/common/src/TruffleConfig.js]: Ah, I commented this before I saw the other hard coded gas prices.

File: packages/common/src/PublicNetworks.js (hunk 2 above)
Comment (nicholaspai) [reply packages/common/src/PublicNetworks.js:52]: I'll remove this gas price for now, let's just let truffle auto detect it

Comment (mrice32) [reply packages/common/src/PublicNetworks.js:52]: It seems like the gas price is still hardcoded for mainnet. IMO, we should make this hardcoded gas price just be the default and allow the --gasPrice to override it in the main config, but definitely open to other ideas/solutions you have!

**Review by mrice32 — COMMENTED** [review]
Just left one comment on the `--gasPrice` parameter thread! Otherwise, this LGTM!

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/common/src/PublicNetworks.js [hunk 4]
```diff
@@ -43,9 +43,21 @@ const PublicNetworks = {
...
+  },
   80001: {
-    name: "mumbai",
-    etherscan: "https://explorer-mumbai.maticvigil.com/"
+    name: "polygon-mumbai",
+    etherscan: "https://explorer-mumbai.maticvigil.com/",
+    customTruffleConfig: {
+      confirmations: 2,
+      timeoutBlocks: 200
+    }
```
Comment (daywiss) [inline packages/common/src/PublicNetworks.js:60]: just to be clear, i think this is ok, but this is more a meta comment about how to organize data for multi chain support. it seems like both chainids and networkids tend to [correlate](https://chainid.network/), and keying only by network id probably is fine even across chains.  I dont know of any instances where 2 chains would have the same network id, but i am paranoid of this.  Do you guys have any opinions on this?

**Review by daywiss — APPROVED** [review]

File: packages/common/src/PublicNetworks.js (hunk 4 above)
Comment (nicholaspai) [reply packages/common/src/PublicNetworks.js:60]: This is a really great point david, we might want to consder storing a chainID in this mapping of networkID's --> network configs, but I don't see anything to do rn

**Pull request merged**